File: cah.server/src/main/java/com/touchgrass/cah/server/model/dto/LobbyDto.java
package com.touchgrass.cah.server.model.dto;

import com.touchgrass.cah.server.model.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.ArrayList;

@AllArgsConstructor
@Getter
@Setter
@ToString
public class LobbyDto {
    private String id;
    private LobbyStatus status;
    private int roundDuration;
    private int maxRoundNumber;
    private int currentRound;
    private Round round;
    private BlackCard currentBlackCard;
    private ArrayList<PlayerDto> playerList;

    public LobbyDto(Lobby lobby) {
        this.id = lobby.getId();
        this.status = lobby.getStatus();
        this.roundDuration = lobby.getRoundDuration();
        this.maxRoundNumber = lobby.getMaxRoundNumber();
        this.currentRound = lobby.getCurrentRound();
        this.currentBlackCard = lobby.getRound().getCurrentBlackCard();
        this.round = lobby.getRound();

        // Convert Player objects to PlayerDto
        this.playerList = new ArrayList<>();
        for (Player player : lobby.getPlayerList().values()) {
            this.playerList.add(new PlayerDto(player));
        }
    }
}

File: caf.app/lib/model/dto/lobby_dto.dart
import 'package:cards_against_humanity/model/dto/black_card_dto.dart';
import 'package:cards_against_humanity/model/dto/round_dto.dart';
import 'package:cards_against_humanity/model/lobby.dart';
import 'package:json_annotation/json_annotation.dart';
import 'player_dto.dart';

part 'lobby_dto.g.dart';

@JsonSerializable()
class LobbyDto {
  LobbyDto(
    this.id,
    this.status,
    this.roundDuration,
    this.maxRoundNumber,
    this.currentRound,
    this.round,
    this.currentBlackCard,
    this.playerList,
  );

  String id;
  LobbyStatus status;
  int roundDuration;
  int maxRoundNumber;
  int currentRound;
  RoundDto? round;
  BlackCardDto? currentBlackCard;
  List<PlayerDto> playerList;

  factory LobbyDto.fromJson(Map<String, dynamic> json) =>
      _$LobbyDtoFromJson(json);

  Map<String, dynamic> toJson() => _$LobbyDtoToJson(this);
}
